src/NHibernate/Logging.cs

```diff
@@ -272,36 +358,158 @@ public bool IsEnabled(InternalLogLevel logLevel)
+
+		// Since v5.1
+		[Obsolete("Please use Fatal(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Fatal(object message, Exception exception) => _this.Fatal(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Debug(object message) => _this.Debug(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Debug(string message) => _this.Debug(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Debug(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Debug(object message, Exception exception) => _this.Debug(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
+		public void DebugFormat(string format, params object[] args) => _this.Debug(format, args);
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(object message) => _this.Info(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(string message) => _this.Info(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Info(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(object message, Exception exception) => _this.Info(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void InfoFormat(string format, params object[] args) => _this.Info(format, args);
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message) => _this.Warn(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(string message) => _this.Warn(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Warn(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message, Exception exception) => _this.Warn(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void WarnFormat(string format, params object[] args) => _this.Warn(format, args);
+	}
+
+	/// <summary>
+	/// Factory building loggers which log nothing.
+	/// </summary>
 	public class NoLoggingLoggerFactory: INHibernateLoggerFactory
```

Comment:
Still a breaking change, I am going to add another commit. It must still implement the old interface too.